### PR TITLE
zfsUnstable: 2.1.6 → 2.1.7-staging

### PR DESCRIPTION
**Pull request by toastal**
With 5.19 kernels now being EoL and my laptop demanding at least 5.17 to function, I want to get ahead of the curve and add support for 6.0 kernels. The 2.1.6 tag was cut so late that it supports 6.0 but isn’t as well tested according to the conversations I read. There aren’t many commits in the `2.1.7-staging` at this point, but this will help keep my system afloat until the real `2.1.7` release. I’ve booted my laptop into this and everything appears to be functioning normally.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by hmenke — APPROVED** [review]

**Comment by hmenke** [comment]
@ofborg test zfs.unstable

File: pkgs/os-specific/linux/zfs/default.nix [hunk 1]
```diff
@@ -227,17 +228,17 @@ in {
...
+    kernelCompatible = kernel.kernelOlder "6.1";
+    latestCompatibleLinuxPackages = linuxPackages_6_0;
 
     # this package should point to a version / git revision compatible with the latest kernel release
     # IMPORTANT: Always use a tagged release candidate or commits from the
     # zfs-<version>-staging branch, because this is tested by the OpenZFS
     # maintainers.
-    version = "2.1.6";
-    # rev = "0000000000000000000000000000000000000000";
+    version = "2.1.7";
```
Comment (hmenke) [inline pkgs/os-specific/linux/zfs/default.nix:238]: Personally I don't mind, because `version` doesn't really have a meaning beyond the store path name, but some people prefer this
```suggestion
    version = "unstable-2021-10-21";
```
or maybe this might also be a good middle ground
```suggestion
    version = "2.1.7-staging-2021-10-21";
```

Comment (toastal) [reply pkgs/os-specific/linux/zfs/default.nix:238]: ah... that name does make sense

**Review by hmenke — APPROVED** [review]

File: pkgs/os-specific/linux/zfs/default.nix [hunk 2]
```diff
@@ -17,6 +17,7 @@
 
 # for determining the latest compatible linuxPackages
 , linuxPackages_5_19 ? pkgs.linuxKernel.packages.linux_5_19
+, linuxPackages_6_0 ? pkgs.linuxKernel.packages.linux_5_20
```
Comment (marijanp) [inline pkgs/os-specific/linux/zfs/default.nix:20]: does it make sense to rename this attribute to `latestCompatibleLinuxPackages`? and get rid of `linuxPackages_5_19`

Comment (hmenke) [reply pkgs/os-specific/linux/zfs/default.nix:20]: Maybe, but a cleanup of the derivation is beyond the scope of this PR.

Comment (toastal) [reply pkgs/os-specific/linux/zfs/default.nix:20]: Good question. I think it would be an improvement, personally.

**Comment by toastal** [comment]
@hmenke I rebased with `master` a couple of times as there were updates that affected kernels. I also included the latest patch added as well as fixing the date year (I put "2021").

File: pkgs/os-specific/linux/zfs/default.nix (hunk 2 above)
Comment (hmenke) [inline pkgs/os-specific/linux/zfs/default.nix:20]: Why did you change this to `linux_5_20`? That attribute doesn't even exist.
```suggestion
, linuxPackages_6_0 ? pkgs.linuxKernel.packages.linux_6_0
```

**Review by hmenke — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/linux/zfs/default.nix (hunk 2 above)
Comment (toastal) [reply pkgs/os-specific/linux/zfs/default.nix:20]: Ouch. Good catch.

Comment (toastal) [reply pkgs/os-specific/linux/zfs/default.nix]: I'm overriding this locally on my machine and *was* testing against 6.0.x. After a reboot:

```console
$ uname -srv                                                                                                                                                                               
Linux 6.0.5 #1-NixOS SMP PREEMPT_DYNAMIC Wed Oct 26 10:53:32 UTC 2022
```

**Review by hmenke — APPROVED** [review]
